wcfsetup/install/files/lib/acp/form/ReactionTypeAddForm.class.php

```diff
@@ -78,26 +84,59 @@ protected function createForm() {
 					->label('wcf.acp.reactionType.isDisabled')
 			]);
 		
+		$this->uploadFormField = UploadFormField::create('iconFile')
+			->label('wcf.acp.reactionType.image')
+			->description('wcf.acp.reactionType.image.description')
+			->required()
+			->maximum(1)
+			->imageOnly(true)
+			->allowSvgImage(true);
+		
 		$iconContainer = FormContainer::create('imageSection')
 			->label('wcf.acp.reactionType.image')
 			->appendChildren([
-				TextFormField::create('iconFile')
-					->label('wcf.acp.reactionType.image')
-					->description('wcf.acp.reactionType.image.description')
-					->required()
-					->addValidator(new FormFieldValidator('invalidPath', function(TextFormField $field) {
-						if (!file_exists(WCF_DIR.'images/reaction/'.$field->getValue())) {
-							$field->addValidationError(new FormFieldValidationError(
-								'invalidPath',
-								'wcf.acp.reactionType.image.invalidPath'
-							));
-						}
-					}))
+				$this->uploadFormField
 			]);
 		
 		$this->form->appendChildren([
 			$dataContainer, 
 			$iconContainer
 		]);
 	}
+	
+	/**
+	 * @inheritDoc
+	 */
+	public function saved() {
+		$this->saveImage($this->objectAction->getReturnValues()['returnValues']);
+		
+		parent::saved();
+	}
+	
+	/**
+	 * Save the image for a reaction type.
+	 * 
+	 * @param       ReactionType    $reactionType
+	 */
+	protected function saveImage(ReactionType $reactionType) {
+		$files = $this->uploadFormField->getValue();
+		
+		/** @var UploadFile $file */
+		$file = array_pop($files);
+		if (!$file->isProcessed()) {
+			$fileName = $reactionType->reactionTypeID . '-'. $file->getFilename();
```

Comment:
Can you confirm that it's impossible to upload files with a `.php` file extension?